service/history/timerQueueTaskExecutorBase.go

```diff
@@ -128,6 +128,13 @@ func (t *timerQueueTaskExecutorBase) executeDeleteHistoryEventTask(
 		return serviceerror.NewInternal("Mutable state has different version but same branch token")
 	}
 
+	if mutableState.IsWorkflowExecutionRunning() {
```

Comment:
I think this method is quite confusing as it will return false for zombie workflows, but zombie workflow may make progress in the future. So I think technically zombie workflow should also return nil here?